src/editors/EditorHeader.jsx

```diff
@@ -1,112 +1,124 @@
+import React, { useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
 import {
-  ActionRow, IconButton, Icon, ModalDialog, Form,
+  ActionRow, Form, IconButton, Icon, ModalDialog,
 } from '@edx/paragon';
-import React, { useContext, useState, useEffect } from 'react';
 import { Edit, Close } from '@edx/paragon/icons';
 import EditorPageContext from './EditorPageContext';
-import { ActionStates } from './data/constants';
+import { ActionStates, mapBlockTypeToName } from './data/constants';
 
 /* The Header component renders based on three states:
     loading -> loaded but not editing -> editing. */
 
-const EditorHeader = () => {
+const EditorHeader = ({ blockType }) => {
   const {
     titleRef, editorRef, setBlockTitle,
     blockValue, blockLoading,
     unitUrl, unitUrlLoading, studioEndpointUrl,
   } = useContext(EditorPageContext);
 
   const [editing, setEditing] = useState(false);
-
-  const [title, setTitle] = useState('');
+  const [title, setTitle] = useState(mapBlockTypeToName(blockType));
   useEffect(() => {
     if (blockLoading === ActionStates.FINISHED) {
-      setTitle(blockValue ? blockValue.data.display_name : '');
+      setTitle(blockValue ? blockValue.data.display_name : mapBlockTypeToName(blockType));
     }
   }, [blockLoading]);
 
-  const handleChange = (e) => {
-    setTitle(e.target.value);
+  const updateTitle = () => {
+    /* This is called when the input field loses focus.
+       It will update the title we are saving and set editing to false. */
+    setBlockTitle(title);
+    setEditing(false);
   };
```

Comment:
I messed around with this logic for a bit, but it makes the most sense to me to save and set editing to false when the input field loses focus.